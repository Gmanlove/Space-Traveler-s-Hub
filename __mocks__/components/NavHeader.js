import { Link } from 'react-router-dom';
import '../styles/NavHeader.css';
import React from 'react';
// import { NavLink } from 'react-router-dom';
// import planetImg from '../public/planet2.png';

const NavHeader = () => {
  const urls = [
    {
      id: 1,
      url: '/rocket',
      text: 'Rocket',
    },
    {
      id: 2,
      url: '/mission',
      text: 'Missions',
    },
    {
      id: 3,
      url: '/Profile',
      text: 'My Profile',
    },
  ];

  return (
    <nav className="navbar">
      <img src="3212580.png" className="PlantImg" alt="Planet-img" />
      <h1 className="nav-heading">Space Travelers Hub</h1>

      <ul className="nav-main">
        {urls.map((url) => (
          <li className="nav-items" key={url.id}>
            <Link to={url.url}>{ url.text }</Link>
          </li>
        ))}
      </ul>
    </nav>
  );
};

export default NavHeader;
