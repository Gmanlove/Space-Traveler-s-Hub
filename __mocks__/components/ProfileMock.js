import React from 'react';
import { useSelector } from 'react-redux';

const ProfileMock = () => {
  const { reserveState } = useSelector(
    (store) => store.rocket,
  );
  const { missionReserve } = useSelector(
    (store) => store.mission,
  );

  return (
    <div className="Profile-holder">
      <h2>My Profile</h2>
      <div className="main">
        <div className="Missions">
          <h4>My Missions</h4>
          <div className="list">
            {missionReserve.map((mission) => (
              <p key={mission.id}>{mission.mission}</p>
            ))}
          </div>
        </div>
        <div className="Rockets">
          <h4>My Rockets</h4>
          <div className="list">
            {reserveState.map((reserved) => (
              <p key={reserved.id}>{reserved.title}</p>
            ))}
          </div>
        </div>
      </div>
    </div>
  );
};
export default ProfileMock;
