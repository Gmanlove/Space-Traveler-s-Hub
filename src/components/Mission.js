import React, { useRef, useState, useEffect } from 'react';
import PropTypes from 'prop-types';
import { useDispatch, useSelector } from 'react-redux';
import { missionButtonReducer } from '../redux/missions/MissionSlice';
import getMission from '../redux/missions/MissionAPI';
import '../styles/Mission.css';

const Mission = ({
  id,
  mission,
  description,
}) => {
  const member = useRef(null);
  const buttonRef = useRef(null);
  const dispatch = useDispatch();
  const [missionText, setMissionText] = useState(
    'Join Mission',
  );
  const [status, setStatus] = useState('NOT A MEMBER');
  const { missionReserve } = useSelector(
    (store) => store.mission,
  );
  useEffect(() => {
    missionReserve.forEach((item) => {
      if (item.id === id) {
        setMissionText('Leave Mission');
        setStatus('ACTIVE MEMBER');
        member.current.classList.add('activeMember');
        buttonRef.current.classList.add('leaveMission');
      }
    });
  }, [id, missionReserve]);

  function setMissionButton(e) {
    if (missionText === 'Join Mission') {
      setMissionText('Leave Mission');
      setStatus('ACTIVE MEMBER');
      member.current.classList.add('activeMember');
      dispatch(
        missionButtonReducer({ id, missionText, mission }),
      );
      e.target.classList.add('leaveMission');
    } else {
      setMissionText('Join Mission');
      setStatus('NOT A MEMBER');
      member.current.classList.remove('activeMember');
      e.target.classList.remove('leaveMission');
      dispatch(
        missionButtonReducer({ id, missionText, mission }),
      );
    }
  }

  return (
    <section className="missionWrapper">
      <div className="data-row">
        <h3>{mission}</h3>
        <p>{description}</p>
        <p ref={member} className="member">{status}</p>
        <button
          ref={buttonRef}
          className="missionButton"
          onClick={(e) => setMissionButton(e)}
          type="button"
        >
          {missionText}
        </button>
      </div>
    </section>
  );
};

Mission.propTypes = {
  id: PropTypes.string.isRequired,
  mission: PropTypes.string.isRequired,
  description: PropTypes.string.isRequired,
};

const MissionPage = () => {
  const dispatch = useDispatch();
  const { content } = useSelector((store) => store.mission);
  useEffect(() => {
    if (content.length < 0) {
      dispatch(getMission());
    }
  }, [dispatch, content]);
  return (
    <>
      <div className="titleRow">
        <h3>Mission</h3>
        <h3>Description</h3>
        <h3>Status</h3>
        <h3>Status</h3>
      </div>

      {content.map((data) => (
        <Mission
          key={data.mission_id}
          id={data.mission_id}
          mission={data.mission_name}
          description={data.description}
          buttonStatus="Join Mission"
        />
      ))}
    </>
  );
};

export default MissionPage;
