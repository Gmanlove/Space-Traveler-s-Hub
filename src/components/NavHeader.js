import React from 'react';
import { NavLink } from 'react-router-dom';
// import planetImg from '../public/planet2.png';

const NavHeader = () => {
  const urls = [
    {
      id: 1,
      url: '/Rocket',
      text: 'Rocket',
    },
    {
      id: 2,
      url: '/missions',
      text: 'Missions',
    },
    {
      id: 3,
      url: '/Profile',
      text: 'My Profile',
    },
  ];

  return (
    <nav className="navbar">
      <img src="3212580.png" className="PlantImg" alt="Planet-img" />
      <h1 className="nav-heading">Space Travelers Hub</h1>

      <ul className="nav-main">
        {urls.map((url) => (
          <li className="nav-items" key={url.id}>
            <NavLink to={url.url}>{ url.text }</NavLink>
          </li>
        ))}
      </ul>
    </nav>
  );
};

export default NavHeader;
