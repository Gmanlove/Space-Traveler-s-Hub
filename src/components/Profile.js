import React from 'react';
import '../styles/Profile.css';

const Profile = () => (
  <div className="Profile-holder">
    <h2>My Profile</h2>
    <div className="main">
      <div className="Missions">
        <h4>My Missions</h4>
        <div className="list">
          <p>Astro-1</p>
          <p>Dawn</p>
          <p>Epoxi</p>
          <p>Fast</p>
          <p>Hubble</p>
        </div>
      </div>
      <div className="Rockets">
        <h4>My Rockets</h4>
        <div className="list">
          <p>New Glenn</p>
          <p>Space Launch System</p>
          <p>Terran 1</p>
          <p>SpaceX Starship</p>
        </div>
      </div>
    </div>
  </div>
);

export default Profile;
