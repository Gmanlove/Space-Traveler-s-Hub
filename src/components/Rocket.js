import '../styles/Rocket.css';
import PropTypes from 'prop-types';
import { useEffect, useState, useRef } from 'react';
import {
  useDispatch,
  useSelector,
} from 'react-redux';
import { buttonReducer } from '../redux/rockets/RocketSlice';
import getRockets from '../redux/rockets/RocketAPI';

const Rocket = ({
  id,
  imgURL,
  title,
  description,
}) => {
  const [buttonText, setButtonText] = useState(
    'Reserve Rocket',
  );
  const { reserveState } = useSelector(
    (store) => store.rocket,
  );
  const buttonRef = useRef(null);

  const dispatch = useDispatch();

  useEffect(() => {
    reserveState.forEach((item) => {
      if (item.id === id) {
        setButtonText('Cancel Reservation');
        buttonRef.current.classList.add(
          'buttonCancel',
        );
      }
    });
  });

  function setButton(e) {
    if (buttonText === 'Reserve Rocket') {
      setButtonText('Cancel Reservation');
      dispatch(
        buttonReducer({ id, buttonText, title }),
      );
      e.target.classList.add('buttonCancel');
    } else {
      setButtonText('Reserve Rocket');
      dispatch(
        buttonReducer({ id, buttonText, title }),
      );
      e.target.classList.remove('buttonCancel');
    }
  }

  return (
    <div className="rocket-component">
      <img src={imgURL} alt="rocket" />
      <span>
        <h3>{title}</h3>
        <p>{description}</p>
        <button
          ref={buttonRef}
          onClick={(e) => setButton(e)}
          className="reserve"
          type="button"
        >
          {buttonText}
        </button>
      </span>
    </div>
  );
};

const RocketPage = () => {
  const dispatch = useDispatch();
  const { content, info } = useSelector(
    (store) => store.rocket,
  );
  useEffect(() => {
    dispatch(getRockets());
  }, [dispatch]);

  if (info === 'success') {
    return content.map((rocket) => (
      <Rocket
        key={rocket.id}
        id={rocket.id}
        imgURL={rocket.flickr_images[1]}
        title={rocket.name}
        description={rocket.description}
      />
    ));
  }
  return (
    <>
      <p>LOADING</p>
    </>
  );
};

Rocket.propTypes = {
  id: PropTypes.string.isRequired,
  imgURL: PropTypes.string.isRequired,
  title: PropTypes.string.isRequired,
  description: PropTypes.string.isRequired,
};

export default RocketPage;
