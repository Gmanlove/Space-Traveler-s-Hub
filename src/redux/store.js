import { configureStore } from '@reduxjs/toolkit';
import { createLogger } from 'redux-logger';
import thunk from 'redux-thunk';
import RocketSlice from './rockets/RocketSlice';
import MissionSlice from './missions/MissionSlice';

const store = configureStore({
  reducer: {
    rocket: RocketSlice,
    mission: MissionSlice,
  },
  middleware: [createLogger(), thunk],
});
export default store;
