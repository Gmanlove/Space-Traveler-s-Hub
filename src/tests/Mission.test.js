import React from 'react';
import {
  fireEvent,
  render,
} from '@testing-library/react';
import { Provider } from 'react-redux';
import { Mission } from '../../__mocks__/components/MissionMock';
import store from '../redux/store';

describe('Mission', () => {
  const component = (
    <Mission
      id="1"
      mission="Moon"
      description="the moon"
    />
  );

  test('should match snapshot', () => {
    expect(component).toMatchSnapshot();
  });
});

describe('Affirming that the status badge changes', () => {
  it('Should modify the badge on click', () => {
    const { container } = render(
      <Provider store={store}>
        <Mission
          id="839ruijkf"
          mission="moon"
          description="to the moon"
        />
      </Provider>,
    );

    const button = container.getElementsByClassName(
      'missionButton',
    )[0];
    const status = container.getElementsByClassName(
      'member',
    )[0];
    fireEvent.click(button);

    expect(status).toHaveTextContent(
      'ACTIVE MEMBER',
    );

    expect(button).toHaveTextContent('Leave Mission');
  });
});
