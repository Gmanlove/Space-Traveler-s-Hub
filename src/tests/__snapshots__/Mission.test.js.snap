// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Mission should match snapshot 1`] = `
<Mission
  description="the moon"
  id="1"
  mission="Moon"
/>
`;
